ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/contract/CodeHash.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.hyperledger.besu.ethereum.core.contract;
+
+import org.hyperledger.besu.datatypes.Hash;
+
+import org.apache.tuweni.bytes.Bytes;
+
+/** key for use by the CodeCache */
+public class CodeHash {
```

Comment:
Maybe you can use the Immutable  library fot that  https://immutables.github.io/